OCaml 5 build failure on MacOS

I tried getting an OCaml 5 on `aarch64-darwin` using this overlay, but encounter a build failure. Any idea whether it's a known issue/limitation with the overlay, or with upstream support for these machines?

Flake:

```nix
{
  description = "foo";

  inputs = {
    nixpkgs.url = "nixpkgs/nixpkgs-unstable";
    flake-utils.url = "github:numtide/flake-utils";

    ocaml-overlay = {
      url = "github:nix-ocaml/nix-overlays";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };

  outputs = { self, nixpkgs, flake-utils, ocaml-overlay }@inputs:
    flake-utils.lib.eachDefaultSystem (system:
      let
        pkgs = import nixpkgs { inherit system; overlays = [ocaml-overlay.overlays.default]; };
      in
        {
          devShell = pkgs.mkShell {
            buildInputs = with pkgs.ocaml-ng.ocamlPackages_5_0; [ ocaml ];
          };
        }
    );
}
```

Failure:
```
error: builder for '/nix/store/hlhq92xxm9kd3n5xfcmgkqxq5qndc9gy-ocaml+flambda-5.0.0.drv' failed with exit code 2;
       last 10 log lines:
       > Error: I/O error: dllunix.so_The boot compiler cannot load DLLs: No such file or directory
       > make[4]: *** [Makefile:166: ../ocamldoc/ocamldoc] Error 2
       > make[4]: Leaving directory '/private/tmp/nix-build-ocaml+flambda-5.0.0.drv-0/ocaml-5.0.0/ocamldoc'
       > make[3]: *** [Makefile:1163: ocamldoc] Error 2
       > make[3]: Leaving directory '/private/tmp/nix-build-ocaml+flambda-5.0.0.drv-0/ocaml-5.0.0'
       > make[2]: *** [Makefile:313: all] Error 2
       > make[2]: Leaving directory '/private/tmp/nix-build-ocaml+flambda-5.0.0.drv-0/ocaml-5.0.0'
       > make[1]: *** [Makefile:328: bootstrap] Error 2
       > make[1]: Leaving directory '/private/tmp/nix-build-ocaml+flambda-5.0.0.drv-0/ocaml-5.0.0'
       > make: *** [/nix/store/0rawfy9g1y4c2h4s60sjm6m80cri92n3-Makefile.nixpkgs:15: nixpkgs_world_bootstrap_world_opt] Error 2
       For full logs, run 'nix log /nix/store/hlhq92xxm9kd3n5xfcmgkqxq5qndc9gy-ocaml+flambda-5.0.0.drv'.
error: 1 dependencies of derivation '/nix/store/pkxbhi6a836hs25m9b2h3xsj90n7s667-nix-shell-env.drv' failed to build
```

Full log: https://gist.github.com/purcell/22c016df4c572d8139f847bbb37326bc

--- a/ocaml/default.nix
+++ b/ocaml/default.nix
@@ -1437,7 +1437,9 @@ with oself;
   });
 
   ocaml = (osuper.ocaml.override { flambdaSupport = true; }).overrideAttrs (_: {
-    enableParallelBuilding = true;
+    buildPhase = ''
+      make defaultentry -j$NIX_BUILD_CORES
+    '';
   });
 
   ocamlformat-lib = osuper.ocamlformat-lib.overrideAttrs (_: {

